### PR TITLE
[Solana]: Support specifying fee payer for an encoded transaction

### PR DESCRIPTION
**Pull request by 10gic**
## Description

Fix issue https://github.com/trustwallet/wallet-core/issues/4114

## How to test

Run Rust, C++, iOS, Android tests

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

File: rust/chains/tw_solana/src/modules/insert_instruction.rs [hunk 1]
```diff
@@ -91,6 +91,30 @@ pub trait InsertInstruction {
         Ok(account_added_at)
     }
 
+    /// Adds a fee payer account to the message.
+    /// Note: The fee payer must NOT be in the account list yet.
+    fn add_fee_payer(&mut self, account: SolanaAddress) -> SigningResult<()> {
+        if self.account_keys_mut().contains(&account) {
+            // For security reasons, we don't allow adding a fee payer if it's already in the account list.
```
Comment (satoshiotomakan) [inline rust/chains/tw_solana/src/modules/insert_instruction.rs:98]: Could you please clarify why don't we need to allow this?

**Review by satoshiotomakan — APPROVED** [review]
Looks good! Only one note - `tw_solana_transaction_add_fee_payer` won't be available to be called from Kotlin, Swift, JS bindings unless we bridge it through `include/TWSolanaTransaction.h`

**Comment by satoshiotomakan** [comment]
Do you have a chance to make a bridge similar to https://github.com/trustwallet/wallet-core/blob/99b69d1c2267354270769f6adef2d370553ce436/src/interface/TWSolanaTransaction.cpp#L29-L41

**Comment by 10gic** [comment]
> Do you have a chance to make a bridge similar to
> 
> https://github.com/trustwallet/wallet-core/blob/99b69d1c2267354270769f6adef2d370553ce436/src/interface/TWSolanaTransaction.cpp#L29-L41

Currently, I haven’t added a getter function for the fee payer. The fee payer is simply the first account in a Solana transaction, and we can easily retrieve it since we already support the Solana transaction decoder.

File: rust/chains/tw_solana/src/modules/insert_instruction.rs (hunk 1 above)
Comment (10gic) [reply rust/chains/tw_solana/src/modules/insert_instruction.rs:98]: I have added some considerations in the comments of the code.

**Comment by 10gic** [comment]
> Looks good! Only one note - `tw_solana_transaction_add_fee_payer` won't be available to be called from Kotlin, Swift, JS bindings unless we bridge it through `include/TWSolanaTransaction.h`

Thanks for the reminder; I will add it later.

**Comment by satoshiotomakan** [comment]
> Currently, I haven’t added a getter function for the fee payer. The fee payer is simply the first account in a Solana transaction, and we can easily retrieve it since we already support the Solana transaction decoder.

No I mean we need to have a `C++ -> Rust` bridge, so it can be used not only from C++ or Rust, but also from Kotlin, Swift etc

**Comment by 10gic** [comment]
Hi @satoshiotomakan, I tried for a while but couldn’t run the test cases locally. Could you please help trigger the CI test cases?

**Review by satoshiotomakan — APPROVED** [review]
Thanks @10gic, let's run it

**Comment by satoshiotomakan** [comment]
@10gic there is a mistake in an iOS test

**Comment by 10gic** [comment]
> @10gic there is a mistake in an iOS test

I have tried to fix it.

**Review by satoshiotomakan — APPROVED** [review]

**Pull request merged**